src/persist/src/postgres.rs

```diff
@@ -0,0 +1,224 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! Implementation of [Consensus] backed by sqlite.
+
+use std::sync::Arc;
+use std::time::Instant;
+
+use async_trait::async_trait;
+use tokio::sync::Mutex;
+use tokio_postgres::{Client, NoTls};
+
+use mz_ore::task;
+
+use crate::error::Error;
+use crate::location::{Consensus, ExternalError, SeqNo, VersionedData};
+
+const SCHEMA: &str = "
+CREATE TABLE IF NOT EXISTS consensus (
+    shard text NOT NULL,
+    sequence_number integer NOT NULL,
+    data bytea NOT NULL,
+    PRIMARY KEY(shard, sequence_number)
+);
+";
+
+/// Configuration to connect to a pg backed implementation of [Consensus].
+#[derive(Debug)]
+pub struct PostgresConsensusConfig {
+    hostname: String,
```

Comment:
any reason not to just make the internals of this a `tokio_postgres::Config`?

Reply:
made it a url string which i felt was even simpler